### PR TITLE
Updating documentation

### PR DESCRIPTION
**Pull request by Cortey**
**Description**
A more cohesive documentation that follows Kyma community guidelines.

**Changes**
- Updated root directory README.md
- Added architecture.md
- Added details.md
- Added development.md
- Updated docs directory README.md

Related issue: #99

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,43 +1,42 @@
...
 
-- Kubernetes cluster
-- Kyma installed on cluster
+- [Kyma](https://kyma-project.io/) cluster 
+- [Kubectl](https://kubernetes.io/docs/tasks/tools/) to deploy the application
 
 ## Installation
 
-1. To run Kyma-Showcase application, you have to clone repository:
+1. To run Kyma Showcase application, clone the repository:
```
Comment (mwisnia97) [inline README.md:16]: Now we have repo name with lowercase, so I think here could be lowercase too.

File: docs/architecture.md [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+8. Depending of the details content a new event may be sent triggering next functions.
+9. Upon clicking a single image in frontend UI a details page is displayed listing all the information recieved from the Google Cloud Platform.
+
+Initial event triggers `Multiple object detection, Landmark detection` and `General labels` functions. The remaining ones get triggered based on the contents of response recieved from Google Cloud Platform.
+
+| Content required | Serverless functions triggered |
+|-----------|-------------|
+| Text/Font (label) | Text detection |
+| Handwriting | Handwriting detection |
+| Logo | Logo detection|
```
Comment (mwisnia97) [inline docs/architecture.md:23]: These lambdas are triggered from the label function, we haven't to add (label) information?

File: docs/development.md [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+To configure Kyma Showcase you need to apply the following parameters in [secrets.yaml](../resources/secrets.yaml) and [frontend-config.yaml](../resources/frontend/frontend-config.yaml).
+
+
+| Parameter | Description |
+|-----------|-------------|
+| GCP_API_KEY | Your Google Cloud Platform API key, for more information on key generation refer to [here](https://cloud.google.com/docs/authentication/api-keys#creating_an_api_key). |
+| GCP_EMAIL | Mandatory credential for Google cloud platform access.|
+| REDIS_PASSWORD | Required as a set up for the authorization layer |
+| API_URL | Defines the API URL that frontend will make calls to. |
```
Comment (mwisnia97) [inline docs/development.md:26]: And API_RULES I think.

File: docs/details.md [hunk 4]
```diff
@@ -0,0 +1,9 @@
+# Details
+
+Project consists of three main components that work together to provide the final result.
+
+[Frontend](./frontend) presents the user with a clickable drag and drop field and a feed of all previously uploaded pictures. After uploading desired image it gets added to the feed in order of upload time. Clicking any image takes the user to a details page where they can read all details info acquired.
+
+[Backend](./backend) manages all calls to the database. Upon sucessfully saving the image backend sends an initial event triggering the serverless functions.
+
+[Functions](./resources/functions) work with the image ID received from the event to acquire the right image from the database and then send it to Google Cloud Platform. Functions update the database entry with newly acquired information and depending on its content can send events triggering more functions in order to get further details from the image.
```
Comment (mwisnia97) [inline docs/details.md:9]: Maybe [Resources] will be better? Then we can write about secrets and subscriptions too.

**Review by mvshao — APPROVED** [review]

**Pull request merged**